source/common/config/grpc_stream.h

```diff
@@ -60,12 +65,13 @@ class GrpcStream : public Grpc::AsyncStreamCallbacks<ResponseProto>,
     }
     stream_ = async_client_->start(service_method_, *this, Http::AsyncClient::StreamOptions());
     if (stream_ == nullptr) {
-      ENVOY_LOG(warn, "Unable to establish new stream");
+      ENVOY_LOG(debug, "Unable to establish new stream to configuration server");
```

Comment:
Where did we land on debug vs. warn on this? I'm concerned we may be losing information here but I'm not positive.

Reply:
Sorry, testing this manually

Reply:
I'm finding in manually testing with Istio I can't trigger this branch (even while setting the discovery address to a dead end), though I do get the error
> 2021-07-20T20:11:11.828138Z	debug	envoy config	StreamAggregatedResources gRPC config stream closed: 14, connection error: desc = "transport: Error while dialing dial tcp: lookup istiod-fail.istio-system.svc on 10.20.0.10:53: no such host"

In my manual testing Istio has an xDS proxy and I wonder if the proxying behavior is preventing this branch from being reached. This causes the last close status to be reset a few lines below and the log level is never escalated. Does this sound expected @howardjohn?

I'll continue with trying to test in a way that causes this branch to be triggered, i.e. without Istio or without the xDS proxy

Reply:
I think one way to trigger this branch for sure is to have the xDS cluster have zero hosts (zero DNS records, etc.). I think it should definitely trigger then.

Reply:
(If time is limited the other option is to just leave it at warn for now.)

Reply:
Potentially `clearCloseStatus` should be moved to [here](https://github.com/envoyproxy/envoy/pull/14616/files/98e6f69e495f94759ab64c2d7a2b5fb8e3f18ea0#diff-4edb4dc54777a58f4899662ad4ac4c5d8caa8544af48214149aa2d46722a1bc1R98)...

Reply:
Sorry I posted my last message before seeing yours. That makes sense, although currently this doesn't work with Istio due to the status always being cleared, so I'd like to fix that before merging

Reply:
Ok, updates:

- Even when there's no host, `onRemoteClose` is triggered, so it currently seems it's fine to keep this log as debug. Example messages:

  >2021-07-21T00:06:41.418077Z	debug	envoy config	StreamAggregatedResources gRPC config stream closed: 14, Cluster not available
...
2021-07-21T00:06:41.938635Z	debug	envoy config	StreamAggregatedResources gRPC config stream closed: 14, no healthy upstream
...
2021-07-21T00:07:12.423970Z	warning	envoy config	StreamAggregatedResources gRPC config stream closed since 31005ms ago: 14, Cluster not available

- To also handle setups where there's a proxy between Envoy and the xDS server, it seems like it makes sense to move `clearCloseStatus` from `establishNewStream` to `onReceiveMessage`, which already contains `backoff_strategy_->reset()`. This is because the xDS proxy needs to accept the stream from Envoy before forwarding it to the xDS server. This seems to also give some robustness in case there are other situations in which Envoy repeatedly establishes a config stream only for it to immediately be closed with `Unavailable`

@mattklein123 WDYT?

Reply:
Sure that sounds fine, thanks.